contracts/state/StateV2.sol

```diff
@@ -0,0 +1,464 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.15;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "./Smt.sol";
+
+interface IVerifier {
+    function verifyProof(
+        uint256[2] memory a,
+        uint256[2][2] memory b,
+        uint256[2] memory c,
+        uint256[4] memory input
+    ) external view returns (bool r);
+}
+
+interface ISmt {
+    function add(uint256 index, uint256 value) external;
+
+    function addHistorical(
+        uint256 _i,
+        uint256 _v,
+        uint64 _timestamp,
+        uint64 _blockNumber
+    ) external;
+
+    function root() external view returns (uint256);
+
+    function getProof(uint256 _index)
+        external
+        view
+        returns (Proof memory);
+
+    function getHistoricalProofByRoot(uint256 _index, uint256 _root)
+    external
+    view
+    returns (Proof memory);
+
+    function getHistoricalProofByBlock(uint256 index, uint64 _block)
+        external
+        view
+        returns (Proof memory);
+
+    function getHistoricalProofByTime(uint256 index, uint64 timestamp)
+        external
+        view
+        returns (Proof memory);
+}
+
+// /**
+//  * @dev Set and get states for each identity
+//  */
+// contract State is Iden3Helpers {
+contract StateV2 is OwnableUpgradeable {
```

Comment:
shouldn't the name of contract be the same?

Reply:
This will not give us any advantages but, on the contrary, bring the mess in the code organisation.
Additionally, from the compiler standpoint we may face compilation name conflict.